x-pack/plugin/logsdb/src/main/java/org/elasticsearch/xpack/logsdb/LogsDBPlugin.java

```diff
@@ -17,23 +17,31 @@
 import java.util.Collection;
 import java.util.List;
 
+import static org.elasticsearch.xpack.cluster.settings.ClusterSettings.CLUSTER_LOGSDB_ENABLED;
 import static org.elasticsearch.xpack.logsdb.SyntheticSourceLicenseService.FALLBACK_SETTING;
 
 public class LogsDBPlugin extends Plugin {
 
     private final Settings settings;
     private final SyntheticSourceLicenseService licenseService;
 
+    private final LogsdbIndexModeSettingsProvider logsdbIndexModeSettingsProvider;
+
     public LogsDBPlugin(Settings settings) {
         this.settings = settings;
         this.licenseService = new SyntheticSourceLicenseService(settings);
+        this.logsdbIndexModeSettingsProvider = new LogsdbIndexModeSettingsProvider(settings);
     }
 
     @Override
     public Collection<?> createComponents(PluginServices services) {
         licenseService.setLicenseState(XPackPlugin.getSharedLicenseState());
         var clusterSettings = services.clusterService().getClusterSettings();
         clusterSettings.addSettingsUpdateConsumer(FALLBACK_SETTING, licenseService::setSyntheticSourceFallback);
+        clusterSettings.addSettingsUpdateConsumer(
+            CLUSTER_LOGSDB_ENABLED,
+            logsdbIndexModeSettingsProvider::updateClusterIndexModeLogsdbEnabled
+        );
         // Nothing to share here:
         return super.createComponents(services);
     }
```

Comment:
Also the `CLUSTER_LOGSDB_ENABLED` setting needs to be returned  by the `getSettings(...)` method? (instead of the stack plugin?)

Reply:
It looks like that if I include the setting here `CUSTER_LOGSDB_ENABLED` I am not able to run javaRestTest and also the CI results in a lot of failures. I still don't know why. For now I will remove it to submit a few more tests and see if they work.

Reply:
So in `SettingsModule` method `registerSetting` we check if we try to register the setting twice.
If we return `CLUSTER_LOGSDB_SETTING` from both `LogsDBPlugin` and `StackPlugin` we end-up registering the same setting twice. I am not sure anyway if deduplicating the list of settings `additionalSettings` in `NodeConstruction` is the right thing to do.

Reply:
If we register the `CLUSTER_LOGSDB_ENABLED` setting in logsdb plugin then we need to remove the registration in the stack module. A setting can't be registered more than once.

This comment is more than a nice to have. I'm ok with keeping `CLUSTER_LOGSDB_ENABLED` in the stack module. We can move this cluster setting in a follow up when we remove the logic in stack module around enabling logsdb index mode in templates.

Reply:
Setting is now only registered in stack module.